snuba/web/query.py

```diff
@@ -47,6 +48,24 @@ def __init__(
         self.meta = meta
 
 
+@dataclass
+class ClickhouseQueryMetadata:
+    request: Request
+    sql: str
+    timer: Timer
+    stats: MutableMapping[str, Any]
```

Comment:
```suggestion
    stats: Mapping[str, Any]
```

It probably makes sense to expose this as immutable (even though it is backed by a mutable data structure) unless you expect/intend to mutate this mapping. I don't think that is the case though?

Reply:
updated